### PR TITLE
feat(RDS): rewrite rds mysql database

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  rewrite rds mysql database
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  rewrite rds mysql database
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rds/' TESTARGS='-run TestAccMysqlDatabase_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccMysqlDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccMysqlDatabase_basic
=== PAUSE TestAccMysqlDatabase_basic
=== CONT  TestAccMysqlDatabase_basic
--- PASS: TestAccMysqlDatabase_basic (1409.99s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       1410.032s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_mysql_database_test.go [hunk 1]
```diff
@@ -7,40 +7,88 @@ import (
...
+		databases := utils.PathSearch("databases", getMysqlDatabaseRespBody, make([]interface{}, 0)).([]interface{})
+		if len(databases) == 0 {
+			break
+		}
+		for _, v := range databases {
+			name := utils.PathSearch("name", v, "").(string)
+			if name == dbName {
+				return v, nil
+			}
+		}
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_mysql_database_test.go]:
```suggestion
database := utils.PathSearch("databases[?name== 'dbname']|[0]", getMysqlDatabaseRespBody)
```
直接可以根据name过滤集合

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_database.go [hunk 2]
```diff
@@ -119,104 +108,184 @@ func resourceRdsDatabaseCreate(ctx context.Context, d *schema.ResourceData, meta
...
+
+		databases := utils.PathSearch("databases", getMysqlDatabaseRespBody,
+			make([]interface{}, 0)).([]interface{})
+		if len(databases) == 0 {
+			break
+		}
+		for _, v := range databases {
+			name := utils.PathSearch("name", v, "").(string)
+			if name == dbName {
+				mErr = multierror.Append(
```
Comment (hmbig2) [inline huaweicloud/services/rds/resource_huaweicloud_rds_mysql_database.go]: separate query and set

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/rds/' TESTARGS='-run TestAccMysqlDatabase_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccMysqlDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccMysqlDatabase_basic
=== PAUSE TestAccMysqlDatabase_basic
=== CONT  TestAccMysqlDatabase_basic
--- PASS: TestAccMysqlDatabase_basic (1281.56s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       1460.461s

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_mysql_database_test.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_mysql_database_test.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_database.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_mysql_database.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_database.go [hunk 3]
```diff
@@ -48,216 +44,257 @@ func ResourceRdsDatabase() *schema.Resource {
...
+
+	getMysqlDatabaseResp, err := pagination.ListAllItems(
+		getMysqlDatabaseClient,
+		"page",
+		getMysqlDatabasePath,
+		&pagination.QueryOpts{MarkerField: ""})
+
 	if err != nil {
-		return common.CheckDeletedDiag(d, err, "error listing RDS db databases")
+		return common.CheckDeletedDiag(d, err, "error retrieving MysqlDatabase")
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_mysql_database.go]:
```suggestion
		return common.CheckDeletedDiag(d, err, "error retrieving RDS Mysql database")
```

Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_mysql_database.go]: fixed

**Pull request merged**